sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveContext.scala

```diff
@@ -34,7 +34,7 @@ class HiveContext private[hive](_sparkSession: SparkSession)
   self =>
 
   def this(sc: SparkContext) = {
-    this(SparkSession.builder().sparkContext(HiveUtils.withHiveExternalCatalog(sc)).getOrCreate())
```

Comment:
`withHiveExternalCatalog` is used only for tests after this merged, so could you move it into a suitable place?

Reply:
> `withHiveExternalCatalog` is used only for tests after this merged, so could you move it into a suitable place?

Done